pkg/client/injection/reconciler/eventing/v1alpha1/broker/stub/controller.go

```diff
@@ -39,14 +41,21 @@ func NewController(
 
 	brokerInformer := broker.Get(ctx)
 
+	classValue := "default" // TODO: update this to the appropriate value.
+	classFilter := reconciler.AnnotationFilterFunc(v1alpha1broker.ClassAnnotationKey, classValue, false /*allowUnset*/)
+
 	// TODO: setup additional informers here.
+	// TODO: remember to use the classFilter from above to filter appropriately.
 
 	r := &Reconciler{}
-	impl := v1alpha1broker.NewImpl(ctx, r)
+	impl := v1alpha1broker.NewImpl(ctx, r, classValue)
 
 	logger.Info("Setting up event handlers.")
 
-	brokerInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+	brokerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: classFilter,
```

Comment:
Yes we can apply filter here. However, there are other places where enqueue happens. We cannot easily apply filter there because we don't have the broker object.

https://github.com/knative/eventing/blob/ba33d59c03ce78f3fa23d3462d3542aef0179588/pkg/reconciler/broker/controller.go#L121-L151

Reply:
but in those cases we _want_ the broker to get enqueued, no?

Reply:
In the case where a trigger is modified, all brokers for all broker classes are enqueued. This is expected because we need the broker object to know if the label is matching this controller. So it enqueues and then the check happens in `Reconcile()` which is a quick no-op and then the reconciler returns with no action.